gef-ui/server/controllers/about-facility/index.js

```diff
@@ -55,30 +57,53 @@ const validateAboutFacility = async (req, res) => {
   let coverEndDate = null;
 
   if (!saveAndReturn) {
+    if (isTrueSet(body.hasBeenIssued)) {
     // Only validate facility name if hasBeenIssued is set to Yes
-    if (isTrueSet(body.hasBeenIssued) && !body.facilityName) {
-      aboutFacilityErrors.push({
-        errRef: 'facilityName',
-        errMsg: `Enter a name for this ${facilityTypeString} facility`,
-      });
-    }
-    if (isTrueSet(body.hasBeenIssued) && !body.shouldCoverStartOnSubmission) {
-      aboutFacilityErrors.push({
-        errRef: 'shouldCoverStartOnSubmission',
-        errMsg: 'Select if you want UKEF cover to start on the day you submit the automatic inclusion notice',
-      });
-    }
-    if ((isTrueSet(body.hasBeenIssued) && body.shouldCoverStartOnSubmission === 'false') && (!coverStartDateDay || !coverStartDateMonth || !coverStartDateYear)) {
-      aboutFacilityErrors.push({
-        errRef: 'coverStartDate',
-        errMsg: 'Enter a cover start date',
-      });
-    }
-    if (isTrueSet(body.hasBeenIssued) && (!coverEndDateDay || !coverEndDateMonth || !coverEndDateYear)) {
-      aboutFacilityErrors.push({
-        errRef: 'coverEndDate',
-        errMsg: 'Enter a cover end date',
-      });
+      if (!body.facilityName) {
+        aboutFacilityErrors.push({
+          errRef: 'facilityName',
+          errMsg: `Enter a name for this ${facilityTypeString} facility`,
+        });
+      }
+      if (!body.shouldCoverStartOnSubmission) {
+        aboutFacilityErrors.push({
+          errRef: 'shouldCoverStartOnSubmission',
+          errMsg: 'Select if you want UKEF cover to start on the day you submit the automatic inclusion notice',
+        });
+      }
+      if (body.shouldCoverStartOnSubmission === 'false' && (!coverStartDateDay || !coverStartDateMonth || !coverStartDateYear)) {
+        aboutFacilityErrors.push({
+          errRef: 'coverStartDate',
+          errMsg: 'Enter a cover start date',
+        });
+      }
+
+      if (body.shouldCoverStartOnSubmission === 'false' && coverStartDateDay && coverStartDateMonth && coverStartDateYear) {
+        const threeMonthsFromNow = add(new Date(), { months: 3 });
+        const startDate = set(new Date(),
+          { year: coverStartDateYear, month: coverStartDateMonth, date: coverStartDateDay });
+
+        if (isPast(startDate)) {
+          aboutFacilityErrors.push({
+            errRef: 'coverStartDate',
+            errMsg: 'Cover start date cannot be before today',
+          });
+        }
+
+        if (isAfter(startDate, threeMonthsFromNow)) {
+          aboutFacilityErrors.push({
+            errRef: 'coverStartDate',
+            errMsg: 'Cover start date cannot be more than 3 months from now',
+          });
+        }
+      }
+
+      if (!coverEndDateDay || !coverEndDateMonth || !coverEndDateYear) {
+        aboutFacilityErrors.push({
+          errRef: 'coverEndDate',
+          errMsg: 'Enter a cover end date',
+        });
+      }
```

Comment:
I think need to add some validation for day/month/year missing, for GDS. i.e "X must include a year": https://design-system.service.gov.uk/components/date-input/#dates-with-errors-example

Reply:
Ive added this